src/partial17.lib/partial17/Partial.h

```diff
@@ -195,13 +193,22 @@ struct Partial {
         return get<index>();
     }
 
+    template<class F>
+    auto visitInitialized(F&& f) {
+        visitInitializedInternal(std::forward<F>(f), m_data.get());
+    }
+    template<class F>
+    auto visitInitialized(F&& f) const {
+        visitInitializedInternal(std::forward<F>(f), m_data.get());
+    }
+
     template<class F>
     auto visitAll(F&& f) {
-        visitInitialized(std::forward<F>(f), m_data.get());
+        visitIndexTypes(f, Indices{});
     }
     template<class F>
     auto visitAll(F&& f) const {
-        visitInitialized(std::forward<F>(f), m_data.get());
+        visitIndexTypes(f, Indices{});

```

Comment:
Please use `std::forward` here. Otherwise the lambdas with captures might not be able to use this.